### ISSUE TITLE
ctd detector detects multiple lines as one area

### ISSUE THREAD

**Issue by JustFrederik** [zyddnys/manga-image-translator#209]
I ran this image once with the ctd detecor and once with the defulat detector. It works fine with the default detector. Every line is detected seperatly and the ocr works fine. The ctd detector on the other hand is drawing the bbox around the whole textbubble. The mask for the text is worse too. The problem is the ocr fails to detect the text with ctd.(ctd detects with max certenty of .3 while with default detector .98)
Original
<img width="250" alt="image" src="https://user-images.githubusercontent.com/121750393/216838067-eaf4dacb-a650-464a-97c9-17bdfb56065a.png">
DEFAULT
<img width="250" alt="default_bboxes_unfiltered" src="https://user-images.githubusercontent.com/121750393/216838232-3562152d-413a-469a-99b3-ace02d13cf54.png">
<img width="250" alt="default_bboxes" src="https://user-images.githubusercontent.com/121750393/216838234-69e4960c-06c0-48fc-9741-70f1e5b1bd2f.png">
<img width="250" alt="default_mask_raw" src="https://user-images.githubusercontent.com/121750393/216838237-545ac90d-5582-430f-9447-85ab1c16e45c.png">
CTD: 
<img width="250" alt="ctd_mask_raw" src="https://user-images.githubusercontent.com/121750393/216838236-9a2aee01-0e41-458b-8d72-f01327074228.png">
<img width="250" alt="ctd_bboxes" src="https://user-images.githubusercontent.com/121750393/216838238-b94d66cf-479e-437d-8694-313849e78303.png">

**Comment by thatDudo** [comment]
I have made the ctd detector use the same textline merge we have use with the default detector (65571e50d23956bc036f78a6ed7b2e71e5d0661b). Please confirm for yourself.

**Comment by JustFrederik** [comment]
it still has the same problem. There have been no changes

**Comment by JustFrederik** [comment]
@thatDudo i think the problem isn’t the merge, but the detection. It fails to detect it as textline. It just detects it as a Textblock and isn’t splitting up the lines.

**Comment by thatDudo** [comment]
My apologies. I did some digging and it seems the error stems directly from the used model which produces following output for the line mask:
![tmp](https://user-images.githubusercontent.com/58403773/222012620-69d529b5-2580-4431-bf91-1430ec46301c.png)
This line mask is split into segments, which explains why the two bboxes are connected.

Here the line mask from the default detector for comparison:
![tmp](https://user-images.githubusercontent.com/58403773/222012818-87d5d0f1-a4d5-43d4-8756-632e0128067c.png)
 
In other words the fault lies in the model and could potentially be fixed through some further post processing or through replacing the line mask of the ctd with the line mask of the default detector though in that case it would seem that the only thing left that maked ctd unique is the textmask...

**Comment by JustFrederik** [comment]
I have an idea how it could be fixed. The problem is that the most mangas are in the vertical format, so the dataset it trained to detect those. When the image is turned by 90˚ it detects both lines seperatly. So somehow it needs to detect if its vertical or horizontal and if its horizontal it needs to rotate the image.

**Comment by thatDudo** [comment]
That is certainly a good idea, though it would require the model to be executed two times and might in some cases also worsen the detection. Something like should have been improved at the training stage by passing in text in both aspect ratios...

**Comment by JustFrederik** [comment]
It could be implemented by a dedicated argument. Its better than nothing and would fix the issue. I cant retrain it because i dont have the original dataset. To improve the result in training would be to create a turned duplicate of every image.

**Comment by thatDudo** [comment]
Youre right. I heard that something like passing rotated images improves stability of the model. Adding an argument would definitely provide a stop gap solution for the ctd detector :thinking:. Maybe we could call that `detection-orientation [auto|h|v]`. I can add that later if you hadnt planned to.
